### DISCUSSION TITLE
How to limit the number of CPUs when generating forecasts with predict()?

### DISCUSSION THREAD

**Discussion by mbrhd** [awslabs/gluonts#1759] [Q&A] [answered]
Hello

The question is in the title, How to limit the number of CPUs when generating forecasts with predict()?

From tests we're doing in my company, it seems that the predict() function takes all the available CPUs to generate the forecasts by default (a sort of multiprocessing?). However, we have other tasks in the same image built to generate the forecasts and we want to limit the number of CPUs.

It seems that the predict() function takes num_workers as parameters but not sure how it works.

Thanks a lot for your help

**Comment by kaleming** [comment, marked as answer]
It's a good question. MXNet/gluon usually takes all available cores.

I have never tried to limit that, but I think you could try some of these options:

1- Use your model inside Docker-container, and limit available cpu on this container.

`docker run --cpuset-cpus="0-2" myapp:latest`

2- Changing some env variable from OpenMP, for instance:


```
os.environ["OMP_NUM_THREADS"] = "1" # export OMP_NUM_THREADS=1
os.environ["OPENBLAS_NUM_THREADS"] = "1" # export OPENBLAS_NUM_THREADS=1
os.environ["MKL_NUM_THREADS"] = "1" # export MKL_NUM_THREADS=1
os.environ["VECLIB_MAXIMUM_THREADS"] = "1" # export VECLIB_MAXIMUM_THREADS=1
os.environ["NUMEXPR_NUM_THREADS"] = "1" # export NUMEXPR_NUM_THREADS=1
```

3- Checking if mxnet contains ENV variables for that:

https://mxnet.apache.org/versions/1.7.0/api/faq/env_var


Good luck

**Reply by mbrhd** [reply]
Thanks a lot @kaleming. Setting up OMP_NUM_THREADS variable in the docker file worked for me.
